src/zenml/integrations/gcp/model_deployers/vertex_model_deployer.py

```diff
@@ -0,0 +1,270 @@
+"""Implementation of the Vertex AI Model Deployer."""
+
+from typing import ClassVar, Dict, Optional, Tuple, Type, cast
+from uuid import UUID
+
+from google.cloud import aiplatform
+
+from zenml.analytics.enums import AnalyticsEvent
+from zenml.analytics.utils import track_handler
+from zenml.client import Client
+from zenml.enums import StackComponentType
+from zenml.integrations.gcp import (
+    VERTEX_SERVICE_ARTIFACT,
+)
+from zenml.integrations.gcp.flavors.vertex_model_deployer_flavor import (
+    VertexModelDeployerConfig,
+    VertexModelDeployerFlavor,
+)
+from zenml.integrations.gcp.google_credentials_mixin import (
+    GoogleCredentialsMixin,
+)
+from zenml.integrations.gcp.model_registries.vertex_model_registry import (
+    VertexAIModelRegistry,
+)
+from zenml.integrations.gcp.services.vertex_deployment import (
+    VertexAIDeploymentConfig,
+    VertexDeploymentService,
+)
+from zenml.logger import get_logger
+from zenml.model_deployers import BaseModelDeployer
+from zenml.model_deployers.base_model_deployer import (
+    DEFAULT_DEPLOYMENT_START_STOP_TIMEOUT,
+    BaseModelDeployerFlavor,
+)
+from zenml.services import BaseService, ServiceConfig
+from zenml.stack.stack import Stack
+from zenml.stack.stack_validator import StackValidator
+
+logger = get_logger(__name__)
+
+
+class VertexModelDeployer(BaseModelDeployer, GoogleCredentialsMixin):
+    """Vertex AI endpoint model deployer."""
+
+    NAME: ClassVar[str] = "Vertex AI"
+    FLAVOR: ClassVar[Type["BaseModelDeployerFlavor"]] = (
+        VertexModelDeployerFlavor
+    )
+
+    @property
+    def config(self) -> VertexModelDeployerConfig:
+        """Returns the `VertexModelDeployerConfig` config.
+
+        Returns:
+            The configuration.
+        """
+        return cast(VertexModelDeployerConfig, self._config)
+
+    def setup_aiplatform(self) -> None:
+        """Setup the Vertex AI platform."""
+        credentials, project_id = self._get_authentication()
+        aiplatform.init(
+            project=project_id,
+            location=self.config.location,
+            credentials=credentials,
+        )
+
+    @property
+    def validator(self) -> Optional[StackValidator]:
+        """Validates that the stack contains a model registry.
+
+        Also validates that the artifact store is not local.
+
+        Returns:
+            A StackValidator instance.
+        """
+
+        def _validate_stack_requirements(stack: "Stack") -> Tuple[bool, str]:
+            """Validates that all the stack components are not local.
+
+            Args:
+                stack: The stack to validate.
+
+            Returns:
+                A tuple of (is_valid, error_message).
+            """
+            # Validate that the container registry is not local.
+            model_registry = stack.model_registry
+            if not model_registry and isinstance(
+                model_registry, VertexAIModelRegistry
+            ):
+                return False, (
```

Comment:
Another question: Why does this model deployer actually need the vertex model registry in the stack? I probably missed it somewhere, but I can't seem to find the connection between the two

Reply:
Ah i will be adding this to the docs, but short answer is we can't deploy in model deployer directly it always need to start with a model id from vertex model registry (we need model to create an endpoint).